core/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -206,8 +207,13 @@ public PluginsService(Settings settings, Path modulesDirectory, Path pluginsDire
                 }
                 Class moduleClass = method.getParameterTypes()[0];
                 if (!Module.class.isAssignableFrom(moduleClass)) {
-                    logger.warn("Plugin: {} implementing onModule by the type is not of Module type {}", pluginEntry.v1().getName(), moduleClass);
-                    continue;
+                    String extraMessage = "";
```

Comment:
Is this extra bit really needed? It's not going to help an end-user at all, just a plugin developer but they can read a breaking changes doc?